### PR TITLE
Replace SPI_execute with SPI_cursor_fetch

### PR DESCRIPTION
**Pull request by zhrt123**
## Problem
Issue: #318 
The result buffer of SPI_execute is limited to 1GB. If the number of rows in `diskquota.table_size` exceeds 3 * 10^7 (500 segments with 60000 tables for example), `SPI_execute("select tableid, size, segid from diskquota.table_size")` will raise an error for `invalid memory alloc request size 1073741824 (context 'SPI TupTable')`.

## Solution

We should fetch table-size entries by portions. Replace SPI_execute() with SPI_cursor_fetch() in load_table_size().

### CODE REVIEW

File: src/gp_activetable.c [hunk 1]
```diff
@@ -936,16 +936,26 @@ get_active_tables_oid(void)
...
-		ereport(ERROR, (errmsg("[diskquota] load_table_size SPI_execute failed: return code %d, error: %m", ret)));
+	if ((plan = SPI_prepare(sql, 0, NULL)) == NULL)
+		ereport(ERROR, (errmsg("[diskquota] SPI_prepare(\"%s\") failed", sql)));
+	if ((portal = SPI_cursor_open(NULL, plan, NULL, NULL, true)) == NULL)
+		ereport(ERROR, (errmsg("[diskquota] SPI_cursor_open(\"%s\") failed", sql)));
+
+	SPI_cursor_fetch(portal, true, 10000);
+
+	if (SPI_tuptable == NULL)
+	{
```
Comment (higuoxing) [inline src/gp_activetable.c:956]: Do we need to call `SPI_cursor_close()` before error out?

**Review by higuoxing — APPROVED** [review]
LGTM with one question.

File: src/gp_activetable.c (hunk 1 above)
Comment (zhrt123) [reply src/gp_activetable.c:956]: Reference to GPDB's code, we do not need to call this function.

**Review by fanfuxiaoran — APPROVED** [review]

**Comment by fanfuxiaoran** [comment]
Better to add some corner test cases for it: such as there are 0 or 10000 , 20000 rows in the table_size

**Comment by zhrt123** [comment]
> Better to add some corner test cases for it: such as there are 0 or 10000 , 20000 rows in the table_size

Since this test will be ver slow, we'd better to test this case at local.

**Pull request merged**